### PR TITLE
GH-14993: [CI][Conda] Fix missing RECIPE_ROOT variable now expected by conda build

### LINKED ISSUE #14993: [CI][Conda] conda-linux-aarch64-* are failing
### Describe the bug, including details regarding any error messages, version, and platform.

https://lists.apache.org/thread/3rqsxwv9oxwwlqr8lfjr9kvpn835ot1g

For example: conda-linux-aarch64-cpu-py3:

https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=41199&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=d9b15392-e4ce-5e4c-0c8c-b69645229181&l=442

```text
+++ mamba create -n sysroot_linux-aarch64 --yes --quiet sysroot_linux-aarch64
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
/opt/conda/bin/cross_compile_support.sh: line 18: RECIPE_ROOT: unbound variable
```

Cc: @h-vetinari 

### Component(s)

Packaging

### PR DESCRIPTION
**Pull request by h-vetinari**
Fixes #14993

* Closes: #14993

### CODE REVIEW

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda

**Comment by h-vetinari** [comment]
So, the R-builds for aarch are still failing with:
```
> library('arrow')
Error: package or namespace load failed for ‘arrow’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/build/r-arrow_1671342405561/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/R/library/arrow/libs/arrow.so':
  /build/r-arrow_1671342405561/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/R/library/arrow/libs/arrow.so: undefined symbol: _ZTIN5arrow17RecordBatchReaderE
Execution halted
```

I don't understand why this is looking for `arrow.so` instead of `libarrow.so` - the recipe hasn't changed.

CC @xhochy

**Comment by h-vetinari** [comment]
> I don't understand why this is looking for `arrow.so` instead of `libarrow.so` - the recipe hasn't changed.

Though on the other hand, the build.sh for r-arrow is calling straight into the top-level `r/`-folder:
```
${R} CMD INSTALL --build r/. ${R_ARGS}
```

so something broke there in the last couple of days. For example, https://github.com/apache/arrow/commit/dca8c07de6089bde7a9c2e10b42b72541a616cb6 changed [`r/configure`](https://github.com/apache/arrow/blame/master/r/configure) quite a bit, which could explain why now the libs are being built differently or not linked correctly anymore.

CC @thisisnic

**Comment by xhochy** [comment]
> I don't understand why this is looking for `arrow.so` instead of `libarrow.so` - the recipe hasn't changed.

This is OK, `arrow.so` is the R module. `libarrow.so` is the pure C++ shared library.

**Comment by xhochy** [comment]
@h-vetinari This looks problematic:

```
2022-12-18T05:51:13.5618933Z *** Found local C++ source: '../cpp'
2022-12-18T05:51:13.5619835Z *** Building libarrow from source
2022-12-18T05:51:13.5620314Z     For build options and troubleshooting, see the install guide:
2022-12-18T05:51:13.5621188Z     https://arrow.apache.org/docs/r/articles/install.html
2022-12-18T05:51:13.8434741Z **** cmake
2022-12-18T05:51:18.7191251Z **** arrow  
2022-12-18T05:51:20.8624297Z **** Error building Arrow C++. Re-run with ARROW_R_DEV=true for debug information. 
2022-12-18T05:51:21.5582869Z PKG_CFLAGS=  -std=c++17
2022-12-18T05:51:21.5589264Z PKG_LIBS= 
```

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-aarch64-cpu-r41

**Comment by kou** [comment]
> For example, https://github.com/apache/arrow/commit/dca8c07de6089bde7a9c2e10b42b72541a616cb6 changed [r/configure](https://github.com/apache/arrow/blame/master/r/configure) quite a bit, which could explain why now the libs are being built differently or not linked correctly anymore.

Oh, the change was by me. I'll look into this.

**Comment by kou** [comment]
@github-actions crossbow submit conda-linux-aarch64-cpu-r41

**Comment by kou** [comment]
@github-actions crossbow submit conda-linux-aarch64-cpu-r41

**Comment by kou** [comment]
@github-actions crossbow submit conda-linux-aarch64-cpu-r41

**Comment by h-vetinari** [comment]
Success! 🥳 

Well done @kou & @xhochy

File: dev/tasks/conda-recipes/r-arrow/build.sh [hunk 1]
```diff
@@ -5,13 +5,14 @@ export DISABLE_AUTOBREW=1
...
 export ARROW_R_CXXFLAGS="${ARROW_R_CXXFLAGS} -std=c++17"
+export LIBARROW_BUILD=false
 
 if [[ "${target_platform}" == osx-* ]]; then
     # See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
     export ARROW_R_CXXFLAGS="${ARROW_R_CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
 fi
 
-export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}"
+export PKG_CONFIG_PATH="${BUILD_PREFIX}/lib/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}"
```
Comment (xhochy) [inline dev/tasks/conda-recipes/r-arrow/build.sh:15]: This doesn't look right. `BUILD_PREFIX` is the build system, not host. It has been failing in the cross-compiling case and this change makes me suspicious whether we then link to the wrong libraries.

Comment (kou) [reply dev/tasks/conda-recipes/r-arrow/build.sh:15]: Ah, sorry. This is not related to fix the R problem.
We may be able remove this line entirely. I'll try it.

**Comment by kou** [comment]
@github-actions crossbow submit conda-linux-*-cpu-r* -g r

**Comment by h-vetinari** [comment]
> `crossbow submit conda-linux-_-cpu-r_ -g r`

is `_` the placeholder here? And what's the difference between before `-g` and after?

Extrapolating from `crossbow submit -g conda`, I would have naïvely expected to be able to run all conda-r builds as
> `crossbow submit - g conda-*-r`

**Comment by kou** [comment]
> is `_` the placeholder here?

`_` isn't a placeholder. The real crossbow command what I used is `@github-actions crossbow submit conda-linux-*-cpu-r* -g r`. `*` is a glob pattern. If the command is processed as a Markdown text, `*...*` is replaced with `<em>...</em>`. This is the trick. :-)

>  And what's the difference between before `-g` and after?

Without `-g`: Find tasks from `tasks:` in `dev/tasks/tasks.yml`

With `-g`: Find groups from `groups:` in dev/tasks/tasks.yml`. Found groups are expanded to tasks.
 
> Extrapolating from `crossbow submit -g conda`, I would have naïvely expected to be able to run all conda-r builds as
> 
> > `crossbow submit - g conda-*-r`

We can use `@github-actions crossbow submit conda-*-r` for them.

**Comment by h-vetinari** [comment]
@kou, I think this would be good to merge (as all the conda builds now pass)? I have other changes lined up, but those would be better in a separate PR IMO.

**Pull request merged**